Description: gpg-agent.socket: Set GPG_AGENT_INFO in systemd for gnupg1
  GPG_AGENT_INFO is already set in debian/Xsession.d/90gpg-agent, but
  that only affects sessions that run Xsession.  This change affects any
  session that imports systemd environment variables.  See:
  https://gitlab.gnome.org/GNOME/gdm/-/commit/448134d3cdbc54e5359ea33d387993b0defdaefa
  https://github.com/sddm/sddm/issues/1276
Author: Richard Hansen <rhansen@rhansen.org>

Forwarded: not-needed

---
--- gnupg2-2.4.7.orig/doc/examples/systemd-user/gpg-agent.socket
+++ gnupg2-2.4.7/doc/examples/systemd-user/gpg-agent.socket
@@ -3,6 +3,10 @@ Description=GnuPG cryptographic agent an
 Documentation=man:gpg-agent(1)
 
 [Socket]
+# gnupg2 doesn't use the GPG_AGENT_INFO environment variable since
+# v2.1.0, but gnupg1 still does so set it in case gnupg1 is installed.
+ExecStartPost=systemctl --user set-environment GPG_AGENT_INFO=%t/gnupg/S.gpg-agent:0:1
+ExecStopPre=systemctl --user unset-environment GPG_AGENT_INFO
 ListenStream=%t/gnupg/S.gpg-agent
 FileDescriptorName=std
 SocketMode=0600
